### PR TITLE
[Aptos Framework][Staking] Simplify with_cap functions

### PR DESCRIPTION
**Pull request by movekevin**
### Description
Currently all _with_cap functions require passing a pool_address, which is redundant as the OwnerCapability object already has pool_address.

### Test Plan
Existing unit + e2e tests.

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/2884)

### CODE REVIEW

File: aptos-move/framework/aptos-framework/sources/stake.move [hunk 1]
```diff
@@ -1643,7 +1603,7 @@ module aptos_framework::stake {
     }
 
     #[test(aptos_framework = @0x1, core_resources = @core_resources, validator = @0x123)]
-    #[expected_failure(abort_code = 0x1000f)]
+    #[expected_failure(abort_code = 0x1000E)]
```
Comment (zekun000) [inline aptos-move/framework/aptos-framework/sources/stake.move:1606]: why this change?

**Review by zekun000 — APPROVED** [review]

File: aptos-move/framework/aptos-framework/sources/stake.move (hunk 1 above)
Comment (movekevin) [reply aptos-move/framework/aptos-framework/sources/stake.move:1606]: The's because I removed one error code and shifted others. After code freeze we should never do that.
